app/controllers/devise/registrations_controller.rb

```diff
@@ -83,9 +83,10 @@ def cancel
   protected
 
   def update_needs_confirmation?(resource, previous)
-    resource.respond_to?(:pending_reconfirmation?) &&
-      resource.pending_reconfirmation? &&
-      previous != resource.unconfirmed_email
+    resource.respond_to?(:can_send_reconfirmation?) &&
```

Comment:
as we are getting a flash notice if we use `pending_reconfirmation?` and toggle is off so I made these changes to make it unique